### PR TITLE
[WIP] Use JSON request service #33

### PR DESCRIPTION
**Pull request by nesterone**
- brute force schema preloading from spec

### CODE REVIEW

**Comment by domoritz** [comment]
Thank you for the PR. The issue with this change is that now the schema validation will be inconsistent between what monaco does and what the external validation does. We should make sure that it's the same.

File: src/components/input-panel/spec-editor/renderer.tsx [hunk 1]
```diff
@@ -183,14 +132,14 @@ class Editor extends React.Component<Props, {}> {
   public componentWillUnmount() {
     document.removeEventListener('keydown', this.handleKeydown);
   }
-  public updateSpec(spec) {
+  public async updateSpec(spec) {
     let parsedMode = this.props.mode;
 
     try {
-      const schema = JSON.parse(spec).$schema;
+      const schemaUrl = JSON.parse(spec).$schema;
```
Comment (domoritz) [inline src/components/input-panel/spec-editor/renderer.tsx]: I wonder whether we can put the logic below somewhere else where we have already parsed the spec into an object.

**Comment by nesterone** [comment]
Looks like I've missed slack channel, joining ...

**Comment by domoritz** [comment]
Having said that, github issues are great because we keep a history.

**Comment by nesterone** [comment]
Yep, let's continue the issue related discussion here
So you are proposing to investigate API of json service in monaco editor for on-demand loading of schemas, I'm not so optimistic but it worse to try

**Comment by domoritz** [comment]
Yes, that would be nice. Vscode uses it so it ought to work but maybe it replies on some node stuff.

**Comment by domoritz** [comment]
Sorry for being so pedantic. Are you still working on this?

**Comment by nesterone** [comment]
Yep, right now I'm looking into the code of https://github.com/Microsoft/vscode-json-languageservice and trying to understand on how it was extracted from VSCode.
There is a big chance that functionality with on-demand loading for schemas was left in VS code base

**Pull request closed without merging**